### ISSUE TITLE
Phone Still ringing after accept call

### ISSUE THREAD

**Issue by amit1576** [BelledonneCommunications/linphone-android#1352]
Hello @Viish ,

I am experiencing a issue while accepting Replaced call.

Steps : 
1) **A** have tried **B**. while **B** was ringing a new replaced call come to **A**. 

2) In code I have called **acceptCall** method in LinphoneMagager class(state == State.IncomingReceived) condition and call get auto answered and I am connected to **C**. [**Like existing code is written we accept call if Auto answer is enabled**]. **B** got auto disconnected [ AS Required].

 else if (state == State.IncomingReceived
                                && (LinphonePreferences.instance().isAutoAnswerEnabled())
                                && !getCallGsmON()) {
                            LinphoneUtils.dispatchOnUIThreadAfter(
                                    new Runnable() {
                                        @Override
                                        public void run() {
                                            if (mCore != null) {
                                                if (mCore.getCallsNb() > 0) {
                                                    mCallManager.acceptCall(call);
                                                }
                                            }
                                        }
                                    },
                                    mPrefs.getAutoAnswerTime());

3) But issue is that while **A** and **C** are talking to each other **A** is also hearing Ring Back tone in Background.

Please suggest where I am making mistake.

[ReplaceCall_PhoneRinging_AfterAccept.txt](https://github.com/BelledonneCommunications/linphone-android/files/6285010/ReplaceCall_PhoneRinging_AfterAccept.txt)

Thanks.

**Comment by Viish** [comment]
From the logs the ringback is correctly stopped:
```
2021-04-09 12:31:37:234 [com.coraltele.coraluc.debug/liblinphone] MESSAGE [ToneManager] [0x6f85cef4a8] state changed : [Ringback, LinphoneCallEnd]
2021-04-09 12:31:37:234 [com.coraltele.coraluc.debug/liblinphone] MESSAGE [ToneManager] stop
2021-04-09 12:31:37:234 [com.coraltele.coraluc.debug/liblinphone] MESSAGE [ToneManager] doStopRingbackTone
```

But maybe this is causing an issue:
```
2021-04-09 12:31:37:213 [com.coraltele.coraluc.debug/liblinphone] ERROR Cannot decline a CallSession that is in state LinphoneCallOutgoingRinging
2021-04-09 12:31:37:213 [com.coraltele.coraluc.debug/liblinphone] MESSAGE Terminate CallSession [0x6f85cef4a8] which is currently in state [LinphoneCallOutgoingRinging]
```
Use `call.terminate()` instead of `call.decline()` to finish your outgoing call.

**Comment by amit1576** [comment]
Hello @Viish ,

Actually I had removed call.decline and call.terminate method as SDK itself doing this. 

2021-04-09 12:31:37:234 [com.coraltele.coraluc.debug/liblinphone] MESSAGE [ToneManager] [0x6f85cef4a8] state changed : [Ringback, LinphoneCallEnd]
2021-04-09 12:31:37:234 [com.coraltele.coraluc.debug/liblinphone] MESSAGE [ToneManager] stop

These above logs are for first call. As there is another logs 

2021-04-09 12:31:38:379 [com.coraltele.coraluc.debug/linphone-android] MESSAGE [Core Manager] Incoming call received, no other call, start ringing

i guess this is causing issue.

Thanks,

**Comment by Viish** [comment]
>  [Core Manager] Incoming call received, no other call, start ringing

This would make you hear the ringtone, not the ringback tone.

**Comment by Viish** [comment]
Try with SDK 5.0.0-alpha-28, I fixed an issue that may cause yours.

**Comment by amit1576** [comment]
> > [Core Manager] Incoming call received, no other call, start ringing
> 
> This would make you hear the ringtone, not the ringback tone.

Yes it is ringtone, I will try with SDK 5.5.0-alpha-28 and let you know.

**Comment by amit1576** [comment]
Hello @Viish ,

Still same issue with **5.0.0-alpha.28**
> Task :app:linphoneSdkSource
Linphone SDK used is 5.0.0-alpha.28+13a5dd6 from local linphone-sdk maven repository2

Please find attached logs.

[PhoneRinging_AfterAccept_NewCall.txt](https://github.com/BelledonneCommunications/linphone-android/files/6322728/PhoneRinging_AfterAccept_NewCall.txt)

**Comment by Viish** [comment]
Thanks, that helps.
Seems the issue is because you accept the call immediately in the call_state_changed callback, so when other callbacks are called after that the state doesn't match anymore.
I'll make a fix and let you know.

**Comment by Viish** [comment]
In the meantime, wait 1 second before answering the call and it should workaround the issue.

**Comment by amit1576** [comment]
Ok i will try.

**Comment by amit1576** [comment]
> In the meantime, wait 1 second before answering the call and it should workaround the issue.

I had tried add 1 second delay but no success. Issue is that when I terminate the old call and accept the new incoming call, the old call get terminated and new call get accepted but in this case the device ring tone started. 

When I tried with accept new incoming call and not terminating old call, the new call get answered and old call get hold but no phone ring toe in this case.

**Comment by Viish** [comment]
You also can disable the ringtone at SDK level and handle it yourself in your own app.

**Comment by amit1576** [comment]
> You also can disable the ringtone at SDK level and handle it yourself in your own app.

Can you suggest which piece of code in SDK.

**Comment by Viish** [comment]
Simply set the following in your factory config file:
```
[sound]
local_ring=
use_native_ringing=0
```

**Comment by Viish** [comment]
Issue should be fixed in SDK 5.0.0-alpha.43+b6de22c
